packages/store-sync/src/postgres/columnTypes.ts

```diff
@@ -0,0 +1,31 @@
+import { customType, PgCustomColumnBuilder } from "drizzle-orm/pg-core";
+import { ColumnBuilderBaseConfig } from "drizzle-orm";
+import superjson from "superjson";
+import { Address, ByteArray, bytesToHex, getAddress, hexToBytes } from "viem";
+
+export const json = <TData>(name: string): PgCustomColumnBuilder<ColumnBuilderBaseConfig & { data: TData }> =>
+  customType<{ data: TData; driverData: string }>({
+    dataType() {
+      // TODO: move to json column type
+      return "text";
+    },
+    toDriver(data: TData): string {
+      return superjson.stringify(data);
+    },
+    fromDriver(driverData: string): TData {
+      return superjson.parse(driverData);
+    },
+  })(name);
+
+export const address = (name: string): PgCustomColumnBuilder<ColumnBuilderBaseConfig & { data: Address }> =>
+  customType<{ data: Address; driverData: ByteArray }>({
+    dataType() {
+      return "bytea";
+    },
+    toDriver(data: Address): ByteArray {
+      return hexToBytes(data);
+    },
+    fromDriver(driverData: ByteArray): Address {
+      return getAddress(bytesToHex(driverData));
+    },
```

Comment:
Any feelings on whether to store hex as binary in Postgres vs hex strings?

Binary will be more compact and probably easier to do some operations, but less ergonomic/human readable when interacting with the DB directly (probably need to do a bunch of casting). This is all abstracted away when working with the DB in TS.

![image](https://github.com/latticexyz/mud/assets/508855/bd520b68-4191-41d1-8456-5f56aad88fc8)

Reply:
I went ahead with bytes/binary columns everywhere as it felt more Postgres-native, even though some tooling (e.g. TablePlus) isn't set up well for displaying it.